components/test_pd/src/server.rs

```diff
@@ -143,14 +143,14 @@ fn hijack_unary<F, R, C: PdMocker>(
             #[allow(clippy::redundant_closure_call)]
             (|| {
                 fail_point!("connect_leader", |_| {
-                    let key = ctx.request_headers().get(0).unwrap();
-                    // The default option has a metadata named "user-agent" so we check the key here.
-                    let v = if key.0 == "pd-forwarded-host" {
-                        std::str::from_utf8(key.1).unwrap()
-                    } else {
-                        ""
-                    };
-                    status = RpcStatus::with_message(RpcStatusCode::UNAVAILABLE, v.to_string());
+                    let details = ctx
```

Comment:
/cc @rleungx Please make sure PD also search for the header instead of just getting the first one.

Reply:
See https://github.com/tikv/pd/blob/841fa61e9710d9622081310e9510c2dd1cbaf8f3/server/grpc_service.go#L1646